Automattic/newspack-popups
# feat: start proof-of-concept for localStorage API

### All Submissions:






### Changes proposed in this Pull Request:



Proof of concept of a `localStorage`-based reader data model + segmentation API.

This proof of concept only explores a potential data model for reader data cached in `localStorage`, and APIs to read/write reader data and match it to prompt segments. It handles reader data storage and retrieval for anonymous readers only, and segmentation criteria for newsletter signup, donation, and article views. Things it doesn't handle (not exhaustive):

* User login and storage to user meta
* Segmentation by user account, favorite categories, or referral URL
* Deprecation of server-based API or AMP elements
* Analytics
* Frequency-based prompt display ("display prompt every _n_ pageviews")
* Scroll-based overlays
* Favorite category updates
* Prompt interaction events (loaded/seen/dismissed)
* Newsletter signup or donation events (upon submitting signup/donation forms)
* Newsletter signup or donation lookup (for existing subscribers/donors)
* Previewing

### How to test the changes in this Pull Request:

1. Create some segments and prompts based on article views, or "not a subscriber"/"not a donor"/"no user account" criteria.
2. Confirm that the prompts display as expected according to the highest-priority matching segment.

### Other information:
